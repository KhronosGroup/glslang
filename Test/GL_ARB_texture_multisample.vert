#version 140
#extension GL_ARB_texture_multisample : enable

out float result;

uniform sampler2DMS data;
uniform sampler2DMSArray data1;
void main()
{
result = texelFetch(data, ivec2(0), 3).r;
ivec2 temp = textureSize(data);
result = texelFetch(data1, ivec3(0), 3).r;
ivec3 temp1 = textureSize(data1);
}
