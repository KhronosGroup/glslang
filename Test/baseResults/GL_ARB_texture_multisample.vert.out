GL_ARB_texture_multisample.vert
Shader version: 140
Requested GL_ARB_texture_multisample
0:? Sequence
0:8  Function Definition: main( ( global void)
0:8    Function Parameters: 
0:10    Sequence
0:10      move second child to first child ( temp float)
0:10        'result' ( smooth out float)
0:10        direct index ( temp float)
0:10          textureFetch ( global 4-component vector of float)
0:10            'data' ( uniform sampler2DMS)
0:10            Constant:
0:10              0 (const int)
0:10              0 (const int)
0:10            Constant:
0:10              3 (const int)
0:10          Constant:
0:10            0 (const int)
0:11      Sequence
0:11        move second child to first child ( temp 2-component vector of int)
0:11          'temp' ( temp 2-component vector of int)
0:11          textureSize ( global 2-component vector of int)
0:11            'data' ( uniform sampler2DMS)
0:12      move second child to first child ( temp float)
0:12        'result' ( smooth out float)
0:12        direct index ( temp float)
0:12          textureFetch ( global 4-component vector of float)
0:12            'data1' ( uniform sampler2DMSArray)
0:12            Constant:
0:12              0 (const int)
0:12              0 (const int)
0:12              0 (const int)
0:12            Constant:
0:12              3 (const int)
0:12          Constant:
0:12            0 (const int)
0:13      Sequence
0:13        move second child to first child ( temp 3-component vector of int)
0:13          'temp1' ( temp 3-component vector of int)
0:13          textureSize ( global 3-component vector of int)
0:13            'data1' ( uniform sampler2DMSArray)
0:?   Linker Objects
0:?     'result' ( smooth out float)
0:?     'data' ( uniform sampler2DMS)
0:?     'data1' ( uniform sampler2DMSArray)
0:?     'gl_VertexID' ( gl_VertexId int VertexId)
0:?     'gl_InstanceID' ( gl_InstanceId int InstanceId)


Linked vertex stage:


Shader version: 140
Requested GL_ARB_texture_multisample
0:? Sequence
0:8  Function Definition: main( ( global void)
0:8    Function Parameters: 
0:10    Sequence
0:10      move second child to first child ( temp float)
0:10        'result' ( smooth out float)
0:10        direct index ( temp float)
0:10          textureFetch ( global 4-component vector of float)
0:10            'data' ( uniform sampler2DMS)
0:10            Constant:
0:10              0 (const int)
0:10              0 (const int)
0:10            Constant:
0:10              3 (const int)
0:10          Constant:
0:10            0 (const int)
0:11      Sequence
0:11        move second child to first child ( temp 2-component vector of int)
0:11          'temp' ( temp 2-component vector of int)
0:11          textureSize ( global 2-component vector of int)
0:11            'data' ( uniform sampler2DMS)
0:12      move second child to first child ( temp float)
0:12        'result' ( smooth out float)
0:12        direct index ( temp float)
0:12          textureFetch ( global 4-component vector of float)
0:12            'data1' ( uniform sampler2DMSArray)
0:12            Constant:
0:12              0 (const int)
0:12              0 (const int)
0:12              0 (const int)
0:12            Constant:
0:12              3 (const int)
0:12          Constant:
0:12            0 (const int)
0:13      Sequence
0:13        move second child to first child ( temp 3-component vector of int)
0:13          'temp1' ( temp 3-component vector of int)
0:13          textureSize ( global 3-component vector of int)
0:13            'data1' ( uniform sampler2DMSArray)
0:?   Linker Objects
0:?     'result' ( smooth out float)
0:?     'data' ( uniform sampler2DMS)
0:?     'data1' ( uniform sampler2DMSArray)
0:?     'gl_VertexID' ( gl_VertexId int VertexId)
0:?     'gl_InstanceID' ( gl_InstanceId int InstanceId)

